nreader.go

```diff
@@ -40,15 +42,23 @@ func (r *nReader) ReadByte() (byte, error) {
 	return buf[0], nil
 }
 
+// ReadN reads n bytes and returns as a byte array of size n.
+// Note: io.Reader can return n,EOF from a read to the end of a file,
+//  even when there was no attempt to go past the end. Because we're
+//  not returning 'bytes read', we will return err=nil in the case
+//  where the requested number of bytes were sucessfully read.
```

Comment:
The "note" part is really an internal concern. Maybe move it to above to EOF handle part inside the func?

It's probably worth simply mentioning that (nil, io.EOF) is returned when the end of the file has been reached here.